### PR TITLE
Add an actual servlet and index.html page to the skeleton

### LINKED ISSUE #16: 403 error with skeleton
mvn archetype:generate -Dappengine-version=1.9.38 -Dapplication-id=hotel -Dfilter=com.google.appengine.archetypes

...

mvn appengine:devserver

leads to a 403 when connecting to http://localhost:8080/

so far I've only tested this on my MacBook.

### PR DESCRIPTION
**Pull request by elharo**
@lesv This enables the skeleton app to run and bring up one servlet and one HTML page. 

fix #16

### CODE REVIEW

File: appengine-skeleton-archetype/src/main/resources/archetype-resources/src/main/webapp/WEB-INF/web.xml [hunk 1]
```diff
@@ -1,7 +1,21 @@
...
+    <servlet-class>${package}.HelloAppEngine</servlet-class>
+  </servlet>
+  <servlet-mapping>
+    <servlet-name>HelloAppEngine</servlet-name>
+    <url-pattern>/hello</url-pattern>
+  </servlet-mapping>
+  <welcome-file-list>
+    <welcome-file>index.html</welcome-file>
+  </welcome-file-list>
+</web-app>
```
Comment (lesv) [inline appengine-skeleton-archetype/src/main/resources/archetype-resources/src/main/webapp/WEB-INF/web.xml]: return needed.

File: appengine-skeleton-archetype/src/main/resources/archetype-resources/src/main/java/HelloAppEngine.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+public class HelloAppEngine extends HttpServlet {
+
+  public void doGet(HttpServletRequest request, HttpServletResponse response) 
+      throws IOException {
+      
+    response.setContentType("text/plain");
+    response.getWriter().println("Hello App Engine!");
+
+  }
+}
```
Comment (lesv) [inline appengine-skeleton-archetype/src/main/resources/archetype-resources/src/main/java/HelloAppEngine.java]: return needed.

**Comment by lesv** [comment]
Two nits.  Assuming there are no objections / additional comments, I'll try to test this out this weekend or early next week and then deploy to maven central.  That usually takes a week.

@tmatsuo may wish to look at this as he wrote the skeleton and the goal wasn't to be a HelloWorld at the time.  Just an empty project. 

@ludoch FYI

File: appengine-skeleton-archetype/src/main/resources/archetype-resources/src/main/java/HelloAppEngine.java (hunk 2 above)
Comment (ludoch) [inline appengine-skeleton-archetype/src/main/resources/archetype-resources/src/main/java/HelloAppEngine.java]: needs @ Override
In general, I would check that all 3 IDEs eclipse, intellij, Netbeans do not emit any warnings or recommendations on all xml/jsp/css/java code emitted by a project template.
All code should also pass the Google style guide.

Comment (ludoch) [inline appengine-skeleton-archetype/src/main/resources/archetype-resources/src/main/java/HelloAppEngine.java]: Why do we need return?

Comment (elharo) [inline appengine-skeleton-archetype/src/main/resources/archetype-resources/src/main/java/HelloAppEngine.java]: done and done

**Comment by elharo** [comment]
If the goal is to have an empty project that does not contain any HTML or Java code, then we could instead make a new archetype that does include minimal Java code as here. As an end user, I often find it helpful to have some sort of basic code to hack on so I can see how everything fits together, rather than starting from a complete blank page.

**Comment by elharo** [comment]
@lesv Ping. If there's a need to have a code-less skeleton archetype, just let me know and I'll convert this into a new HelloWorld archetype instead.

**Comment by lesv** [comment]
Yep - there was nothing from either @ludoch or @tmatsuo so I'll try to do this today.  I don't like changing the basic functionality (as I didn't create this).  We'll have to do a new bunch for Flex at some point.

File: appengine-skeleton-archetype/src/main/resources/archetype-resources/src/main/java/HelloAppEngine.java (hunk 2 above)
Comment (lesv) [inline appengine-skeleton-archetype/src/main/resources/archetype-resources/src/main/java/HelloAppEngine.java]: Still needs a return at End of Line - see the red mark on line 38.  -- I can't edit the PR to fix for you. :(

File: appengine-skeleton-archetype/src/main/resources/archetype-resources/src/main/webapp/index.html [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    <table>
+      <tr>
+        <td colspan="2" style="font-weight:bold;">Available Servlets:</td>        
+      </tr>
+      <tr>
+        <td><a href='/hello'>The servlet</a></td>
+      </tr>
+    </table>
+  </body>
+</html>
```
Comment (lesv) [inline appengine-skeleton-archetype/src/main/resources/archetype-resources/src/main/webapp/index.html]: See red mark line 22 - please add newline.

**Comment by lesv** [comment]
I'm Ok w/ this -- even though it changes things a bit.  I still want newlines added before I merge, however.

**Comment by elharo** [comment]
newlines added

**Comment by rdayal** [comment]
Can this be merged?

**Comment by lesv** [comment]
Will publish today.

**Comment by lesv** [comment]
Did you look at: https://cloud.google.com/appengine/docs/java/tools/maven#choosing_an_archetype

**Comment by lesv** [comment]
I'm going to wait on your other PR #25 to merge before I publish.

**Comment by lesv** [comment]
I have one more PR - I would like to wait on before publishing.

**Comment by lesv** [comment]
It has been released to Sonotype, you should be able to find it on search.maven.com in the AM.  It will likely be Monday before you see it from the command line.  (They run archetypes on the weekend).

**Comment by elharo** [comment]
Cool, thanks.

On Wed, Jun 15, 2016 at 10:45 PM, Les Vogel notifications@github.com
wrote:

> It has been released to Sonotype, you should be able to find it on
> search.maven.com in the AM. It will likely be Monday before you see it
> from the command line. (They run archetypes on the weekend).
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/GoogleCloudPlatform/appengine-maven-archetypes-java/pull/23#issuecomment-226374650,
> or mute the thread
> https://github.com/notifications/unsubscribe/AA9X6JfdB6qPl3ypELIVSA1mCl64nYkeks5qMLjYgaJpZM4ItDew
> .

## 

Elliotte Rusty Harold
elharo@macfaq.com

**Pull request merged**